server.go

```diff
@@ -133,52 +133,121 @@ func (app *Yarana) replyText(replyToken, text string) error {
 	return nil
 }
 
-func (app *Yarana) handleText(message *linebot.TextMessage, replyToken string, source *linebot.EventSource) error {
-	// Handle user input
-	switch message.Text {
+func (app *Yarana) handleText(message *linebot.TextMessage, replyToken string, source *linebot.EventSource) (err error) {
+	// Analyze text message
+	userReq := NewUserTextRequest()
+	err = userReq.AnalyzeInputText(message.Text)
+	if err != nil {
+		app.sorryAndShowHelp(replyToken) // TODO: Do we have to manange replyToken? because we can use replyToken only once in a request.
+		return nil                       // regard invalid text for parsing as no error
 	}
+	switch userReq.Type {
+	case RequstTypeGetKotos:
+		err = app.processGetKotos(replyToken, source.UserID, userReq.VariableKeyword)
+		if err != nil {
+			return err
+		}
+	case RequstTypeAddKoto:
+		err = app.processAddKoto(replyToken, source.UserID, userReq.VariableKeyword)
+		if err != nil {
+			return err
+		}
+	case RequstTypeGetActivities:
+		err = app.processGetActivities(replyToken, source.UserID, userReq.VariableKeyword)
+		if err != nil {
+			return err
+		}
+	case RequstTypeAddActivity:
+		err = app.processAddActivity(replyToken, source.UserID, userReq.VariableKeyword)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
+func (app *Yarana) handleImage(message *linebot.ImageMessage, replyToken string) error {
+	return nil
+}
+
+func (app *Yarana) processGetKotos(replyToken string, userID string, keyword string) error {
 	// Get Kotos
-	kotos, err := app.dataCall.GetKotosByUserID("d59964bb713fd6f4f5ef6a7c7e029387") // sample data in Cosmos DB
+	kotos, err := app.dataCall.GetKotosByUserID(userID)
 	if err != nil {
 		return err
 	}
 
-	// Add Koto
-	kotoToAdd, _ := NewKotoData("", source.UserID, "Demo AddKoto")
-	err = app.dataCall.AddKoto(kotoToAdd)
+	// Make text to send
+	var textToSend string
+	if len(kotos) == 0 || kotos == nil {
+		textToSend = "No Koto Data"
+	} else {
+		textToSend = kotos[0].Title
+	}
+	if _, err := app.bot.ReplyMessage(
+		replyToken,
+		linebot.NewTextMessage(textToSend),
+	).Do(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (app *Yarana) processAddKoto(replyToken string, userID string, keyword string) error {
+	kotoToAdd, _ := NewKotoData("", userID, keyword)
+	errChan := make(chan error, 1)
+
+	// Add Koto Data
+	go func() {
+		err := app.dataCall.AddKoto(kotoToAdd)
+		errChan <- err
+	}()
+
+	var textToSend string
+
+	err := <-errChan
 	if err != nil {
+		textToSend = "I'm sorry I failed to add your new やること."
+		if _, err := app.bot.ReplyMessage(
+			replyToken,
+			linebot.NewTextMessage(textToSend),
+		).Do(); err != nil {
+			return err
+		}
+		return err // TODO: I wonder the `err` scope. app.bot.ReplyMessage error or app.dataCall.AddKoto error?
```

Comment:
Please investigate it.

Reply:
OK.

Reply:
I confirmed the returned `err` is `app.dataCall.AddKoto`'s one.
`err` at line:211 is in "if sentence scope".

Here is evidence of that.
https://play.golang.org/p/mtbZ3Cdirs7